x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/dashboards/api/utils/installation.ts

```diff
@@ -0,0 +1,167 @@
+  savedObjectsClient,
+  request,
+  context,
+}: InstallTranslatedProps): Promise<number> => {
+  const dashboardMigrationsClient = securitySolutionContext.siemMigrations.getDashboardsClient();
+
+  let installedCount = 0;
+  const installationErrors: Error[] = [];
+
+  // Get installable dashboard migrations
+  const dashboardBatches = await dashboardMigrationsClient.data.dashboards.searchBatches(
+    migrationId,
+    { ids, installable: true }
+  );
+
+  let dashboardsToInstall = await dashboardBatches.next();
+  while (dashboardsToInstall.length) {
+    const { dashboardsToUpdate, errors } = await installDashboards(
+      dashboardsToInstall,
+      securitySolutionContext,
+      contentManagement,
+      savedObjectsClient,
+      request,
+      context
+    );
+    installedCount += dashboardsToUpdate.length;
+    installationErrors.push(...errors);
+    await dashboardMigrationsClient.data.dashboards.update(dashboardsToUpdate);
+    dashboardsToInstall = await dashboardBatches.next();
+  }
+
+  // Throw an error if needed
+  if (installationErrors.length) {
+    throw new Error(installationErrors.map((err) => err.message).join(', '));
+  }
+
+  return installedCount;
+};
+
+const installDashboards = async (
+  dashboardsToInstall: StoredDashboardMigration[],
+  securitySolutionContext: SecuritySolutionApiRequestHandlerContext,
+  contentManagement: ContentManagementServerSetup,
+  savedObjectsClient: SavedObjectsClientContract,
+  request: KibanaRequest,
+  context: RequestHandlerContext
+): Promise<{
+  dashboardsToUpdate: Array<{ id: string; elastic_dashboard: { id: string } }>;
+  errors: Error[];
+}> => {
+  const errors: Error[] = [];
+  const dashboardsToUpdate: Array<{ id: string; elastic_dashboard: { id: string } }> = [];
+
+  const createDashboardsOutcome = await initPromisePool({
+    concurrency: MAX_DASHBOARDS_TO_CREATE_IN_PARALLEL,
+    items: dashboardsToInstall,
+    executor: async (dashboard) => {
+      try {
+        // Parse the dashboard data (assuming it's JSON)
+        const dashboardData = JSON.parse(dashboard.original_dashboard.data);
+
+        // Create dashboard using content management client
+        const client = contentManagement.contentClient
+          .getForRequest({
+            request,
+            requestHandlerContext: context,
+          })
+          .for<SavedObject>(CONTENT_ID, LATEST_VERSION);
+
+        const { result } = await client.create(
+          {
+            title: dashboard.original_dashboard.title,
+            description: dashboard.original_dashboard.description,
+            // Add other dashboard attributes as needed
+            ...dashboardData,
+          },
+          {
+            id: undefined, // Let the system generate an ID
+            references: [],
+            initialNamespaces: [],
+          }
+        );
+
+        dashboardsToUpdate.push({
+          id: dashboard.id,
+          elastic_dashboard: {
+            id: result.item?.id ?? '',
```

Comment:
Can we keep the `id` as _undefined_ or _null_ when we have an empty `id`? We use an _exist_ query on the `id` field to know if things are installed or not, an empty string `''` would make the dashboard be considered as installed.

```suggestion
            id: result.item?.id || null,
```